libr/core/rtr.c

```diff
@@ -1188,15 +1196,41 @@ static void r_rap_packet_fill(ut8 *buf, const ut8* src, int len) {
 	}
 }
 
+static void r_core_rtr_rap_run(RCore *core, const char *input) {
+	r_core_cmdf (core, "o rap://%s", input);
+}
+
+static int r_core_rtr_rap_thread (RThread *th) {
+	if (!th) return false;
+	RapThread *rt = th->user;
+	if (!rt || !rt->core) return false;
+	r_core_rtr_rap_run (rt->core, rt->input);
+	return true;
+}
+
 R_API void r_core_rtr_cmd(RCore *core, const char *input) {
 	char bufw[1024], bufr[8], *cmd_output = NULL;
 	const char *cmd = NULL;
 	int i, cmd_len, fd = atoi (input);
 
 	if (*input==':' && !strchr (input + 1, ':')) {
-		r_core_cmdf (core, "o rap://%s", input);
+		r_core_rtr_rap_run (core, input);
+		return;
+	}
+
+	if (*input=='&') {
+		if (rapthread) {
+			eprintf ("RAP Thread is already running\n");
+			eprintf ("This is experimental and probably buggy. Use at your own risk\n");
+		} else {
+			RapThread rt = { core, input + 1 };
+			rapthread = r_th_new (r_core_rtr_rap_thread, &rt, 0);
+			r_th_start (rapthread, 1);
```

Comment:
maybe use true/false instead of 0,1 here, the signature of the function must be changed (I will do that later)